source/blog/_posts/2018-01-12-mutation-testing-with-infection.md

```diff
@@ -0,0 +1,374 @@
+
+## Installation and Integration
+I will use <a href="https://github.com/teiling88/mutation-testing/commit/78dfccf89d8b97040f5efbffa9542b18fed44c59">this</a> commit as the starting point for the next steps. If you execute infection the first time, it will guide you through a small wizard to configure infection for your application. It is not necessary to keep a special configuration command in mind. Foremost the wizard wants to configure your source directory:
+
+```bash
+Welcome to the Infection config generator
+
+We did not find configuration file. The following questions will help us to generate it for you.
+
+What source directories do you want to include (comma separated)? [src]: 
+  [0] .
+  [1] build
+  [2] src
+  [3] tests
+  [4] vendor
+ > 
+```
+After that, you can exclude folders in your source directory:
+
+```bash
+There can be situations when you want to exclude some folders from generating mutants.
+You can use glob pattern (*Bundle/**/*/Tests) for them or just regular dir path.
+It should be relative to the source directory.
+Press <return> to stop/skip adding dirs.
+
+Any directories to exclude from within your source directories?: 
+```
+Next step is to configure the timeout for each test:
+
+```bash
+Single test suite timeout in seconds [10]: 
+```
+
+At last we have to configure a path for your infection log file:
+
+```
+Where do you want to store the text log file? [infection-log.txt]: build/infection.log
+```
+After successfully configuration Infection is executed for the first time:
+
+```bash
+Configuration file "infection.json.dist" was created.
+
+    ____      ____          __  _
+   /  _/___  / __/__  _____/ /_(_)___  ____ 
+   / // __ \/ /_/ _ \/ ___/ __/ / __ \/ __ \
+ _/ // / / / __/  __/ /__/ /_/ / /_/ / / / /
+/___/_/ /_/_/  \___/\___/\__/_/\____/_/ /_/
+ 
+    0 [>---------------------------] < 1 secRunning initial test suite...
+
+Phpunit version: 5.7.2
+
+   12 [============================] < 1 sec
+
+Generate mutants...
+
+Processing source code files: 1/1
+Creating mutated files and processes: 8/8
+.: killed, M: escaped, S: uncovered, E: fatal error, T: timed out
+
+........                                             (8 / 8)
+
+8 mutations were generated:
+       8 mutants were killed
+       0 mutants were not covered by tests
+       0 covered mutants were not detected
+       0 errors were encountered
+       0 time outs were encountered
+
+Metrics:
+         Mutation Score Indicator (MSI): 100%
+         Mutation Code Coverage: 100%
+         Covered Code MSI: 100%
+
+Please note that some mutants will inevitably be harmless (i.e. false positives).
+```
+
+The result page looks very similar like humbug result page. Infection created 8 mutations which all were killed. Finally the initial configuration process is easy and the generated `infection.json.dist` file is ready for more detailed settings.
```

Comment:
The result page looks very similar **to the** humbug result page.

Reply:
fixed